source/server/listener_manager_impl.cc

```diff
@@ -373,6 +373,24 @@ ListenerManagerStats ListenerManagerImpl::generateStats(Stats::Scope& scope) {
 
 bool ListenerManagerImpl::addOrUpdateListener(const envoy::config::listener::v3::Listener& config,
                                               const std::string& version_info, bool added_via_api) {
+  std::string name;
+  if (!config.name().empty()) {
+    name = config.name();
+  } else {
+    name = server_.api().randomGenerator().uuid();
```

Comment:
Is a random uuid helpful for users as it will be used in the exception below?

Reply:
Actually, this code is moved from https://github.com/envoyproxy/envoy/pull/22005/files#diff-f7639f0659fc4c081810d0a87d2e9e1aa9dc1c1423ea2d683ae491c4479e95f1L402

But yes, you are right, it isn't good for the user in the log or exception. I think there are two ways to improve that:
1. just asking user should specify a name
2. we log both the name and addresses.

If we choose the second one, we can do that in the separated PR, since the whole listener is full of logs with only the name, those are out of scope for this PR.

Reply:
Both ok to me, as a user I do not want a random uuid.

Reply:
got it